### PR TITLE
Add JS backtraces to Python exceptions.

### PR DESCRIPTION
**Pull request by PetterS**
Pretty important because otherwise debugging JS errors is harder than it has to be.

Simply converting the JS  stack object to a string gives us an indented string over multiple lines, so we don't have to do anything more complicated.

The check `if (!JS_IsNull(exception) && !JS_IsUndefined(exception))` guards against what I think is a QuickJS bug (it segfaults and is reported).

### CODE REVIEW

File: module.c [hunk 1]
```diff
@@ -59,6 +59,15 @@ static void teardown_time_limit(ContextData *context) {
 	}
 }
 
+// Converts a JSValue to a C string.
+// Can return NULL. Deallocate with JS_FreeCString.
+static const char* js_to_cstring(JSContext* context, JSValue value) {
+	JSValue string = JS_ToString(context, value);
```
Comment (simonlindholm3) [inline module.c]: Test for JS_EXCEPTION

File: module.c [hunk 2]
```diff
@@ -198,23 +207,31 @@ static PyObject *quickjs_to_python(ContextData *context_obj, JSValue value) {
 	} else if (tag == JS_TAG_EXCEPTION) {
 		// We have a Javascript exception. We convert it to a Python exception via a C string.
 		JSValue exception = JS_GetException(context);
-		JSValue error_string = JS_ToString(context, exception);
-		const char *cstring = JS_ToCString(context, error_string);
+		const char *cstring = js_to_cstring(context, exception);
+		const char* stack_cstring = NULL;
+		if (!JS_IsNull(exception) && !JS_IsUndefined(exception)) {
+			JSValue stack = JS_GetPropertyStr(context, exception, "stack");
```
Comment (simonlindholm3) [inline module.c]: test for JS_EXCEPTION

File: module.c (hunk 1 above)
Comment (PetterS) [reply module.c]: OK. (This function returned NULL for exceptions before as well)

File: module.c (hunk 2 above)
Comment (PetterS) [reply module.c]: Done.

File: module.c (hunk 1 above)
Comment (simonlindholm3) [reply module.c]: Sorry, I didn't mean the check the input argument for being an exception (I think that's the caller's responsibility, exception checking should be done just after calling an API, exception values should not be passed around at all ideally), but rather `string` being an exception. Stringification can fail, it calls toString for instance.

Comment (PetterS) [reply module.c]: Makes perfect sense.

**Pull request merged**